uaclient/contract.py

```diff
@@ -263,7 +262,7 @@ def request_updated_contract(
             )
         except util.UrlError as e:
             if isinstance(e, ContractAPIError):
-                if API_ERROR_INVALID_TOKEN in e:
+                if hasattr(e, "code") and e.code == 401:
```

Comment:
Is 400 a straight invalid (too many characters or something?) vs expired vs anything else?

Reply:
yes 400 is bad request, so I don't think we should treat that as invalid token error. It could be client doesn't know how to talk to a new contract server